WordPress/src/main/java/org/wordpress/android/ui/posts/services/PostUploadNotifier.java

```diff
@@ -137,18 +137,19 @@ public void updateNotificationSuccess(PostModel post, SiteModel site, boolean is
         notificationBuilder.setStyle(new NotificationCompat.BigTextStyle().bigText(message));
         notificationBuilder.setAutoCancel(true);
 
+        long notificationId = getNotificationIdForPost(post);
         // Tap notification intent (open the post list)
         Intent notificationIntent = new Intent(mContext, PostsListActivity.class);
         notificationIntent.addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP);
         notificationIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         notificationIntent.putExtra(WordPress.SITE, site);
         notificationIntent.putExtra(PostsListActivity.EXTRA_VIEW_PAGES, post.isPage());
-        PendingIntent pendingIntentPost = PendingIntent.getActivity(mContext, 0,
-                notificationIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+        PendingIntent pendingIntentPost = PendingIntent.getActivity(mContext,
+                (int)notificationId,
```

Comment:
Good catch! In this case we just need some number that is different than the rest of possible numbers that we are using, and given it is not likely that anyone is going to upload more than 2 posts at the very same time, it's easier to just use a combination of the siteId plus postId (which in practice are always numbers way lower than 65535).
So yes, theoretically it can happen, but practically it won't :)